main.nf

```diff
@@ -301,9 +301,11 @@ ${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style
  * Parse software version numbers
  */
 process get_software_versions {
+    publishDir "${params.outdir}/pipeline_info", mode: 'copy'
```

Comment:
Looking good! Final piece of the puzzle is to add some logic here. We don't want to save the `software_versions_mqc.yaml` file to the results directory, as this is an intermediate file just for use with MultiQC. So, we want to tell nextflow to only save `software_versions.csv`..

```suggestion
    publishDir "${params.outdir}/pipeline_info", mode: 'copy',
        saveAs: {filename ->
            if (filename.indexOf(".csv") > 0) filename
            else null
        }
```

This checks if the filename contains the string `.csv` (`indexOf` finds the position of a substring in a string, so if it returns a number greater than 0, it contains that substring). If it is found, it's saved as `filename` (the original filename). If not, it's saved as `null` which skips the file.